Content.Client/_Goobstation/Fishing/Overlays/FishingOverlay.cs

```diff
@@ -0,0 +1,137 @@
+
+    public override OverlaySpace Space => OverlaySpace.WorldSpaceBelowFOV;
+
+    public FishingOverlay(IEntityManager entManager, IPlayerManager player)
+    {
+        _entManager = entManager;
+        _player = player;
+        _transform = _entManager.EntitySysManager.GetEntitySystem<SharedTransformSystem>();
+        _progressColor = _entManager.System<ProgressColorSystem>();
+
+        // Load the progress bar texture
+        var sprite = new SpriteSpecifier.Rsi(new("/Textures/_Goobstation/Interface/Misc/fish_bar.rsi"), "icon");
+        _barTexture = _entManager.EntitySysManager.GetEntitySystem<SpriteSystem>().Frame0(sprite);
+    }
+
+    protected override void Draw(in OverlayDrawArgs args)
+    {
+        var handle = args.WorldHandle;
+        var rotation = args.Viewport.Eye?.Rotation ?? Angle.Zero;
+        var xformQuery = _entManager.GetEntityQuery<TransformComponent>();
+
+        const float scale = 1f;
+        var scaleMatrix = Matrix3Helpers.CreateScale(new Vector2(scale, scale));
+        var rotationMatrix = Matrix3Helpers.CreateRotation(-rotation);
+
+        // Define bounds for culling entities outside the viewport
+        var bounds = args.WorldAABB.Enlarged(5f);
+        var localEnt = _player.LocalSession?.AttachedEntity;
+
+        // Calculate the size of the texture in world units
+        var textureSize = new Vector2(_barTexture.Width, _barTexture.Height) / EyeManager.PixelsPerMeter;
+
+        var scaledTextureSize = textureSize * BarScale;
+
+        // Define the progress bar's width as a fraction of the texture width
+        var barWidth = scaledTextureSize.X * BarWidthFraction;
+
+        // Iterate through all entities with ActiveFisherComponent
+        var enumerator = _entManager.AllEntityQueryEnumerator<ActiveFisherComponent, SpriteComponent, TransformComponent>();
+        while (enumerator.MoveNext(out var uid, out var comp, out var sprite, out var xform))
+        {
+            // Skip if the entity is not on the current map, has invalid progress, or is not the local player
+            if (xform.MapID != args.MapId ||
+                comp.TotalProgress == null ||
+                comp.TotalProgress < 0 ||
+                uid != localEnt)
+                continue;
+
+            // Get the world position of the entity
+            var worldPosition = _transform.GetWorldPosition(xform, xformQuery);
+            if (!bounds.Contains(worldPosition))
+                continue;
+
+            // Set up the transformation matrix for rendering
+            var worldMatrix = Matrix3Helpers.CreateTranslation(worldPosition);
+            var scaledWorld = Matrix3x2.Multiply(scaleMatrix, worldMatrix);
+            var matty = Matrix3x2.Multiply(rotationMatrix, scaledWorld);
+            handle.SetTransform(matty);
+
+            // Calculate the position of the progress bar relative to the entity
+            var position = new Vector2(
+                sprite.Bounds.Width / 2f,
```

Comment:
`SpriteComponent.Bounds' is obsolete: 'Use SpriteSystem.GetLocalBounds() instead.`

```suggestion
                _sprite.GetLocalBounds((uid, sprite)).Width / 2f,
```